addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -82,7 +84,7 @@ options.registry.TableOfContent = options.Class.extend({
         const headingsEls = this.$target.find(this.targetedElements).toArray()
             .filter(el => !el.closest('.o_snippet_desktop_invisible'));
         const areHeadingsEqual = this.oldHeadingsEls.length === headingsEls.length
-            && this.oldHeadingsEls.every((el, i) => el.isEqualNode(headingsEls[i]));
+            && this.oldHeadingsEls.every((el, i) => el.isEqualNode(headingsEls[i]) && el.closest('[data-visibility-id]'));
```

Comment:
I don't understand this change, how is this related to knowing if the 2 heading DOM are equals?

Reply:
Hello @rdeodoo,

Current situation is here that if the content of the navbar before the change of the DOM is equal to the content of the navbar after the change of the DOM, then there is no need to regenerate the navbar.I've encountered a situation where I need to add the `data-visibility-id` attribute to `<a>` tag. However, in a scenario where the `oldHeadingsEls` and `headingsEls` are the same, it won't generate the `<a>` tag again, and consequently, the `data-visibility-id` attribute value won't be included. Currently, I'm exploring how the code behaves in this condition.